Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -2526,6 +2549,12 @@ void WebProcessProxy::addAllowedFirstPartyForCookies(const WebCore::RegistrableD
     send(Messages::WebProcess::AddAllowedFirstPartyForCookies(firstPartyDomain), 0);
 }
 
+void WebProcessProxy::releaseMemoryAfterSuspensionTimerFired()
+{
+    auto activity = throttler().backgroundActivity("Release memory after suspension"_s);
+    send(Messages::WebProcess::ReleaseMemory({ }), 0);
```

Comment:
The activity goes out of scope right away and thus the process likely doesn't stay awake long enough to process the IPC.

Reply:
Also, we don't want to wake up suspended processes in general. This will page them in memory again.

Reply:
Ah, good catch! Will fix in upcoming patch. In this case, perhaps it would be ok to wake up the process, since we will only run a limited amount of code to free memory, which should not touch many pages?

Reply:
I don't think we can do that. It WILL page a lot of stuff in and increase jetsam priority.
 Feel free to talk to @bnham about it too. I remember he was opposed to doing this for the reasons I mentioned.